peerid.go

```diff
@@ -7,12 +7,14 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-type PeerId ecdsa.PublicKey
+type PeerId [64]byte
 
-func (p PeerId) ToBytes() []byte {
-	if p.X == nil || p.Y == nil {
+func PublicKeyToPeerID(k ecdsa.PublicKey) *PeerId {
```

Comment:
I don't think the pointer here is necessary. 
```suggestion
func PublicKeyToPeerID(k ecdsa.PublicKey) PeerId {
```